catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/CachedResourceMetacardComparator.java

```diff
@@ -108,11 +83,7 @@ public static boolean isSame(
       return false;
     }
 
-    if (!allMetacardMethodsReturnMatchingAttributes(cachedMetacard, updatedMetacard)) {
-      return false;
-    }
-
-    return allMetacardAttributesEqual(cachedMetacard, updatedMetacard);
+    return allMetacardMethodsReturnMatchingAttributes(cachedMetacard, updatedMetacard);
```

Comment:
❓ Think we can improve how we do the checksum comparison here and return `true` right away when both checksums are equal and not `null`? https://github.com/codice/ddf/blob/f5fcbae9a884cc5c8191dd77cd67474390b65974/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/CachedResourceMetacardComparator.java#L80-L82

Reply:
note for future after our conversation: using checksum as a fail fast and not success because algorithm possibilities. extra checks on modified and created to safeguard.